src/Features/Lsif/Generator/Generator.cs

```diff
@@ -331,6 +351,39 @@ void MarkImplementationOfSymbol(ISymbol baseMember)
             return documentVertex.GetId();
         }
 
+        private static async Task GenerateSemanticTokensAsync(
+            Document document,
+            ILsifJsonWriter lsifJsonWriter,
+            IdFactory idFactory,
+            SourceText text,
+            LsifDocument documentVertex)
+        {
+            var ids = new Dictionary<string, int>();
+
+            // Create lookup table for semantic token types.
+            for (var i = 0; i < SemanticTokenTypes.AllTypes.Count; i++)
+            {
+                ids[SemanticTokenTypes.AllTypes[i]] = i;
+            }
+
+            // Compute colorization data.
+            var data = await SemanticTokensHelpers.ComputeSemanticTokensDataAsync(
+                document,
+                ids,
+                new LspProtocol.Range()
+                {
+                    Start = new Position(0, 0),
+                    End = new Position(text.Lines.Count - 1, 0)
```

Comment:
if the last line contains text that should probably be included as well (aka extend past character 0).